class.phpmailer.php

```diff
@@ -1474,8 +1496,8 @@ protected function smtpSend($header, $body)
         }
 
         // Attempt to send to all recipients
-        foreach (array($this->to, $this->cc, $this->bcc) as $togroup) {
-            foreach ($togroup as $to) {
+        if (!empty($this->envelope_to)) {
+            foreach ($this->envelope_to as $to) {
```

Comment:
> foreach($headers['from'] as $from) {
> $mail->setFrom($from['address'], $from['name']);
> }

There can be only one sender. Fix your code to have a scaler in $headers['from'], or arbitrary select the first item in array $headers['from'].